client/client.go

```diff
@@ -365,23 +445,47 @@ func (c *client) handleDispatcher(loopCtx context.Context, dc string, tsoDispatc
 		// we will try to create the stream first.
 		if stream == nil {
 			if needUpdate {
-				err = c.updateLeader()
+				err = c.updateMember()
 				if err != nil {
 					select {
 					case <-loopCtx.Done():
 						return
 					default:
 					}
-					log.Error("[pd] failed updateLeader", errs.ZapError(err))
+					log.Error("[pd] failed update member", errs.ZapError(err))
 					continue
 				}
 				needUpdate = false
 			}
-			ctx, cancel = context.WithCancel(loopCtx)
-			done := make(chan struct{})
-			go c.checkStreamTimeout(ctx, cancel, done)
-			stream, err = pdpb.NewPDClient(c.getClientConnByDCLocation(dc)).Tso(ctx)
-			done <- struct{}{}
+			cc, url := c.getClientConnByDCLocation(dc)
+			cctx, cancel = context.WithCancel(loopCtx)
+			stream, err = c.createTsoStream(cctx, cancel, pdpb.NewPDClient(cc))
+			failpoint.Inject("unreachableNetwork", func() {
+				cancel()
+				stream, cancel = nil, nil
+				err = status.New(codes.Unavailable, "unavailable").Err()
+			})
+			rpcErr, ok := status.FromError(err)
+			// encounter the network error
+			if ok && isNetworkError(rpcErr.Code()) {
```

Comment:
When ` status.FromError(err)` return `nil, true`, will line 470 panic?

Reply:
`ok` will prevent the nil problem.